blueocean-dashboard/src/main/js/components/Pipelines.jsx

```diff
@@ -18,7 +18,12 @@ export default class Pipelines extends Component {
         }
 
         const pipelineRecords = pipelines
-            .map(data => new PipelineRecord(data))
+            .map(data =>
+                'numberOfFolders' in data ?
+                    new FolderRecord(data) :
+                    new PipelineRecord(data)
+            )
+            .filter(data => !(data instanceof FolderRecord))
```

Comment:
yeah - I think that may be better and it maps to the data closer. 

The day may come where we need a folder type, but I don't think like this.